paper.md

```diff
@@ -0,0 +1,129 @@
+   index: 2
+
+date: 11 October 2023
+bibliography: paper.bib
+---
+
+# Summary
+
+Given a set of observations that each belong to a certain class, supervised classification aims to learn a classification model that can predict the class of a new, unlabeled observation [@Cunningham:2008]. This modeling process finds extensive application in real-life scenarios, including but not limited to medical diagnostics, recommendation systems, credit scoring, and sentiment analysis.
+
+In various real-world scenarios, such as those pertaining to the detection of particular conditions like fraud, faults, pollution, or rare diseases, a severe discrepancy between the number of observations in each class can occur. This is known as class imbalance. This poses a problem if assumptions inherent in the classification model imply hindered performance when the model is trained on imbalanced data as is commonly the case [@Ali:2015]. Two prevalent strategies for mitigating class imbalance, when it poses a problem to the classification model, involve either increasing the representation of data in the smaller classes through oversampling or reducing the data in the larger classes through undersampling. It may be also possible to achieve even greater performance by combining both approaches [@Zeng:2016] or by resampling the data multiple times and training the classification model on each resampled dataset to form an ensemble model that aggregates results from different model instances [@Liu:2009].
+
+## Imbalance.jl
+
+In this work, we present, `Imbalance.jl`, a software toolbox implemented in the Julia programming language that offers over 10 well established techniques that help address the class imbalance issue. Additionally, we present a companion package, `MLJBalancing.jl`, which facilitates the integration of resampling methods with classification models, to create a seamless machine learning pipeline that behaves like a single unified model and implements a general version of the EasyEnsemble algorithm presented in [@Liu:2009]. The set of resampling techniques implemented in `Imbalance.jl` and `MLJBalancing.jl` are shown \autoref{techniques}. Although no combination resampling techniques are explicitly presented, they are easy to form using the `BalancedModel` wrapper found in `MLJBalancing.jl`.
+
+
+The package offers a pure functional interface for each method implemented. An arbitrary `Imbalance.jl` resampling method `resample` can be used in the following fashion:
+
+```julia
+X_after, y_after = resample(X, y)
+```
+Here `X_after, y_after` are `X, y` after resampling.
+
+A `ratios` hyperparameter is almost always present to control the degree of oversampling or undersampling to be done for each class. All hyperparameters for a resampling method have default values. To avoid using the defaults, an arbitrary `Imbalance.jl` resampling method `resample` that also has other hyperparameters `a`, `b` and `c` can be used in the following fashion:
```

Comment:
```suggestion
A `ratios` hyperparameter is almost always present to control the degree of oversampling or undersampling to be done for each class. All hyperparameters for a resampling method have default values. To override defaults, an arbitrary `Imbalance.jl` resampling method `resample` that also has other hyperparameters `a`, `b` and `c` can be used in the following fashion:
```

It took me a few moments to parse the meaning of  "an arbitrary `Imbalance.jl` resampling method `resample`". I suggest we be more concrete. Something like "Defaults can be overriden as shown in the following example..." and give code that can be immediately run.

Reply:
How about if I just fix the punctuation as follows "An arbitrary Imbalance.jl resampling method, resample, can be used in the following fashion:". I would have gone with your suggestion but I use this same sentence in the paragraph before this one and it's not as easy to substitute there.

Reply:
On second thoughts, is this example even necessary? This seems like  "user manual" stuff, and the manual and docstrings are already so good...

Reply:
No, but I have seen papers like that of imbalanced-learn and others give a small intro to package syntax as well. Have no problem to remove them if you wish but I feel they won't hurt.

Reply:
I don't think it helps much, given my first criticism. My (revised) suggestion is use a concrete example of a resampling function, and concrete hyperparameter names, or just drop this example -  it's too abstract for quick reading. A software announcement is for skim reading.

Reply:
Alrighty. I will go with the revised suggestion.